build.sbt

```diff
@@ -1147,7 +1147,8 @@ lazy val `scio-smb`: Project = project
       "com.google.protobuf" % "protobuf-java" % protobufVersion,
       "joda-time" % "joda-time" % jodaTimeVersion,
       "org.apache.beam" % "beam-vendor-guava-26_0-jre" % beamVendorVersion,
-      "org.slf4j" % "slf4j-api" % slf4jVersion
+      "org.slf4j" % "slf4j-api" % slf4jVersion,
+      "com.github.ben-manes.caffeine" % "caffeine" % caffeineVersion
```

Comment:
This can be "provided" maybe?

Reply:
> Is there a way to validate the keys in `PreKeyed` sink matches metadata? Just for extra sanity check.

yeah I added a sampling transform that checks! Users can opt out of it if they're very confident/don't want to incur the cost of the extra transform: https://github.com/spotify/scio/pull/3079/files#diff-677cd9b1b47a87b82312e6df2ae18092R768 Unfortunately the Write transform doesn't depend on that check but the job will still fail.